src/main/java/it/pagopa/transactions/utils/EcsEncoderLogMasker.java

```diff
@@ -29,13 +36,34 @@ public byte[] encode(ILoggingEvent event) {
 
     private String maskMessage(String message) {
         StringBuilder sb = new StringBuilder(message);
+        ArrayList<List<Integer>> safeIndexes = new ArrayList<>();
```

Comment:
An idea to avoid add safe pattern processing (to skip log timestamps) could be the following: 
here we are masking the whole log message that is in json format
(ex: `{"@timestamp":"2023-06-23T07:22:20.892Z","log.level": "INFO","message":"Transaction initialized for rptId: 77777777777311116723749670106","ecs.version": "1.2.0","service.name":"pagopa-ecommerce-transactions-service","service.version":"0.19.13","service.environment": │
│`). 
Maybe here can be worth to mask only the message field content so that the timestamp exclusion can be skipped altogether as done here https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/4f5658bc3ecd5cc3d072227385ba969b9643484f/src/test/java/it/pagopa/transactions/utils/EcsEncoderLogMaskerTests.java#L58-L60.

So masking, then, can be applied to the log message field only masking that field with the configured regex leaving other log fields intact.

What do you think?

Reply:
It's a good idea, i'm investigating to find a way to manipulate the JsonObject

Reply:
@pietro-tota I did as you suggested!